### PR TITLE
TRIFFHIR-386: Fixed issue with changing IDs for codesystems and value…

### PR DESCRIPTION
**Pull request by ajagann1**
…sets.

### CODE REVIEW

**Comment by seanmcilvenna** [comment]
This seems to have broken the unit tests in the CI build: https://travis-ci.org/lantanagroup/trifolia-on-fhir/builds/725336630?utm_source=slack&utm_medium=notification Please take a look and update.

**Comment by ajagann1** [comment]
Fixed. Should be good to look at now

File: apps/client/src/app/shared/code-system.service.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ export class CodeSystemService {
 
   }
 
-    public save(codeSystem: CodeSystem): Observable<CodeSystem> {
-        if (codeSystem.id) {
+    public save(codeSystem: CodeSystem, id?: String): Observable<CodeSystem> {
+        if (id) {
```
Comment (seanmcilvenna) [inline apps/client/src/app/shared/code-system.service.ts:15]: What is the value of the `id` argument here? We don't do anything with it other than check its existance in `if (id) {`...

File: apps/client/src/app/codesystem/codesystem.component.ts [hunk 2]
```diff
@@ -158,7 +158,7 @@ export class CodesystemComponent extends BaseComponent implements OnInit, OnDest
       return;
     }
 
-    this.codeSystemService.save(this.codeSystem)
+    this.codeSystemService.save(this.codeSystem, this.getCodeSystemID())
```
Comment (seanmcilvenna) [inline apps/client/src/app/codesystem/codesystem.component.ts:161]: I'm thinking the changes to `codesystem.component.ts` and `codesystem.service.ts` are not necessary... Can you confirm?

**Comment by seanmcilvenna** [comment]
Summary of review:
* The changes to `fhir.controller.ts` seem good!
* I'm not sure the changes to the other two files are needed. Please confirm.

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]

File: apps/client/src/app/codesystem/codesystem.component.ts (hunk 2 above)
Comment (ajagann1) [reply apps/client/src/app/codesystem/codesystem.component.ts:161]: The issue here is that the codesystem resource doesn't store the ID within it's data object.  When it goes through save and does its check for the ID, it was completely erroring out and properly saving the codesystem. This fix corrects that.

File: apps/client/src/app/shared/code-system.service.ts (hunk 1 above)
Comment (ajagann1) [reply apps/client/src/app/shared/code-system.service.ts:15]: Same as above.

**Comment by ajagann1** [comment]
Hey @seanmcilvenna,

I think this is working now.  Please feel free to take a look and let me know what you think of the changes.

Thanks!

File: apps/client/src/app/shared/code-system.service.ts (hunk 1 above)
Comment (ajagann1) [reply apps/client/src/app/shared/code-system.service.ts]: Removed

File: apps/client/src/app/codesystem/codesystem.component.ts (hunk 2 above)
Comment (ajagann1) [reply apps/client/src/app/codesystem/codesystem.component.ts]: Removed

**Pull request merged**